cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java

```diff
@@ -0,0 +1,344 @@
+        }
+
+        // get the manifest from balo file
+        Path baloFilePath = packageBaloFile.get();
+        final BaloProject baloProject = BaloProject.loadProject(baloFilePath, null);
+        final String orgName = baloProject.currentPackage().packageOrg().toString();
+
+        // Validate the org-name
+        if (!RepoUtils.validateOrg(orgName)) {
+            throw createLauncherException(
+                    "invalid organization name provided \'" + orgName
+                            + "\'. Only lowercase alphanumerics "
+                            + "and underscores are allowed in an organization name and the maximum "
+                            + "length is 256 characters");
+        }
+
+        // Validate the pkg-name
+        if (!RepoUtils.validatePkg(pkgName)) {
+            throw createLauncherException("invalid package name provided \'" + pkgName + "\'. Only "
+                    + "alphanumerics, underscores and periods are allowed in a module name "
+                    + "and the maximum length is 256 characters");
+        }
+
+        // todo: need to add after ballerina.toml changes
+        // check if there are any dependencies with balo path
+//        List<String> dependenciesWithBaloPath = baloProject.currentPackage().packageDescriptor().dependencies().stream()
+//                .filter(dep -> dep.getMetadata().getPath() != null).map(Dependency::getModuleID)
+//                .collect(Collectors.toList());
+//
+//        if (!dependenciesWithBaloPath.isEmpty()) {
+//            throw createLauncherException("dependencies cannot be given by path when pushing module(s) to "
+//                    + "remote. check dependencies in Ballerina.toml: [" + String.join(", ", dependenciesWithBaloPath)
+//                    + "]");
+//        }
+
+        // check if the package is already there in remote repository
+        PackageDescriptor.Dependency pkgAsDependency = new PackageDescriptor.Dependency(
+                baloProject.currentPackage().packageName(),
+                baloProject.currentPackage().packageOrg(),
+                baloProject.currentPackage().packageVersion());
+
+        if (isDependencyAvailableInRemote(pkgAsDependency)) {
+            throw createLauncherException(
+                    "package '" + pkgAsDependency.toString() + "' already exists in " + "remote repository("
+                            + getRemoteRepoURL() + "). build and push after "
+                            + "updating the version in the Ballerina.toml.");
+        }
+
+        return baloProject;
+    }
+
+    /**
+     * Push balos to remote repository in the order of there dependencies are resolved.
+     *
+     * @param balos The remaining balos to be pushed.
+     * @throws IOException When trying to access remote repository
+     */
+    private static void recursivelyPushBalos(Map<Path, List<PackageDescriptor.Dependency>> balos) throws IOException {
+        // if there are no more balos to push.
```

Comment:
I guess we dont need this now since package has all the modules

Reply:
removed